crates/stages/src/stages/headers.rs

```diff
@@ -92,6 +95,9 @@ impl<DB: Database, D: HeaderDownloader, C: Consensus, H: HeadersClient, S: Statu
         while let Some(headers) = stream.next().await {
             match headers.into_iter().collect::<Result<Vec<_>, _>>() {
                 Ok(res) => {
+                    // Register elapsed time of request
+                    histogram!("stages.headers.request_time", start.elapsed());
+                    counter!("stages.headers.counter", res.len() as u64);
```

Comment:
we probably want these metrics, but it would be nicer to extract this to a separate call, so it doesn't clutter this function here.